client/src/model/values/instantaneousValues/scalarValues/AbstractScalarValue.hx

```diff
@@ -0,0 +1,39 @@
+package model.values.instantaneousValues.scalarValues;
+import model.values.instantaneousValues.InstantaneousValueI;
+import model.values.instantaneousValues.vectorValues.VectorValue;
+import model.values.instantaneousValues.vectorValues.VectorValueI;
+
+/**
+ * ...
+ * @author AdvaitTrivedi
+ */
+class AbstractScalarValue implements ScalarValueI 
+{
+	/* INTERFACE model.values.instantaneousValues.scalarValues.ScalarValueI */
+	
+	public function toVectorValue():VectorValueI 
+	{
+		var vector = new VectorValue();
+		vector.push(this);
+		return vector;
+	}
+	
+	public function toString():String { return ""; }
```

Comment:
For an "abstract method" in haxe, you can call Assert.assert(false); before the return.  Then, at least we get a run time error if the method is not overridden.